org.eclipse.draw2d.tests/src/org/eclipse/draw2d/test/AdvancedGraphicsTests.java

```diff
@@ -77,15 +76,15 @@ public void paintControl(PaintEvent e) {
 		shell.setBounds(100, 100, 800, 600);
 		shell.open();
 		Display d = shell.getDisplay();
-		d.timerExec(PREVIEW_DELAY, new Runnable() {
+		d.asyncExec(new Runnable() {
 			public void run() {
 				if (!shell.isDisposed())
 					shell.close();
 			}
 		});
 		while (!shell.isDisposed())
 			while (!d.readAndDispatch())
-				d.sleep();
+				; // do nothing...
```

Comment:
I'd like to keep the timeout in the Maven config, if possible. Even if we solve the deadlock for this test, it might pop up in another test and then we're stuck again for 6h.

_However_, I think I can replace the second loop with calling `waitEventLoop`, given that I've already moved that to the BaseTestCase class, which explicitly asks for a timeout.

Reply:
> I'd like to keep the timeout in the Maven config, if possible. Even if we solve the deadlock for this test, it might pop up in another test and then we're stuck again for 6h.

Yes this is a good point. sow e should keep the timeout in maven.